clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1521,6 +1522,25 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
   // Emit the standard function prologue.
   StartFunction(GD, ResTy, Fn, FnInfo, Args, Loc, BodyRange.getBegin());
 
+  SyclOptReportHandler &OptReportHandler = CGM.getDiags().OptReportHandler;
+  if (OptReportHandler.HasOptReportInfo(FD)) {
```

Comment:
If the `Optional` bit works out from above, this would become:
```
if (Optional<SmallVectorImpl<OptReportInfo>&> ORI = OptReportHandler.GetInfo(FD))
```